tmail-backend/jmap/extensions-opensearch/src/main/java/com/linagora/tmail/james/jmap/OSEmailAddressContactSearchEngine.java

```diff
@@ -160,7 +181,7 @@ public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String p
             .get()
             .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.hits().hits()))
             .map(Throwing.function(this::extractContentFromHit).sneakyThrow())
-            .distinct(contact -> contact.fields().address());
+            .distinct(emailAddressContact -> emailAddressContact.fields().identifier());
```

Comment:
This is related to a previously fixed issue: 
- https://github.com/linagora/tmail-backend/issues/541

I generate the `identifier` using both the email address and full name.  
As a result, the frontend may return multiple contacts with the same email address but different full names when searching.